packages/nimble/R/BUGS_macros.R

```diff
@@ -0,0 +1,126 @@
+
+#' EXPERIMENTAL: Turn a function into a model macro builder
+#'
+#' A model macro expands one line of code in a nimbleModel into one or more new lines.  This supports compact programming by defining re-usable modules.  \code{model_macro_builder} takes as input a function that constructs new lines of model code from the original line of code.  It returns a function suitable for internal use by \code{nimbleModel} that arranges arguments for input function.  Macros are an experimental feature and are available only after setting \code{nimbleOptions(enableModelMacros = TRUE)}.
```

Comment:
nit: To ease the job of github reviewers, please limit line length to 80 characters. Emacs even has a macro to wrap long lines into paragraphs.